### PR TITLE
Change `Container`'s `LayoutParams` based on `RecyclerView` direction

### PR DESCRIPTION
**Pull request by veyndan**

### CODE REVIEW

File: redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt [hunk 1]
```diff
@@ -99,7 +99,14 @@ internal open class ViewLazyList(context: Context) : LazyList<View> {
 
   final override val placeholder = Placeholders(recyclerView.recycledViewPool)
 
-  private val linearLayoutManager = LinearLayoutManager(recyclerView.context)
+  private val linearLayoutManager = object : LinearLayoutManager(recyclerView.context) {
+    // Identical to the implementation of [LinearLayout.generateDefaultLayoutParams].
```
Comment (veyndan) [inline redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt:103]: With a name like `LinearLayoutManager`, one would expect that at least the implementation of `generateDefaultLayoutParams()` to be the same as `LinearLayout`'s.

File: redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt [hunk 2]
```diff
@@ -99,7 +99,14 @@ internal open class ViewLazyList(context: Context) : LazyList<View> {
...
+    override fun generateDefaultLayoutParams(): RecyclerView.LayoutParams? = when (orientation) {
+      RecyclerView.HORIZONTAL -> RecyclerView.LayoutParams(WRAP_CONTENT, WRAP_CONTENT)
```
Comment (JakeWharton) [inline redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt:105]: Wrap, match?

**Review by JakeWharton — APPROVED** [review]

File: redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt (hunk 2 above)
Comment (veyndan) [reply redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt:105]: In `LinearLayout` [they use `WRAP_CONTENT` for both directions](https://developer.android.com/reference/android/widget/LinearLayout#generateDefaultLayoutParams()) when the orientation is horizontal. We definitely don't have to match what they do. I just want it to be the least surprising default.

Comment (JakeWharton) [reply redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt:105]: Yeah. It seems weird to be asymmetrical, but I don't really have a strong opinion on what is the right thing to do. I suspect most important is to be consistent across all of our platforms.

**Pull request merged**